### PR TITLE
Fix Issue 15630 - gdb: <incomplete type> for enums values

### PR DESCRIPTION
**Pull request by JinShil**
Attempted revival of #6464

Special thanks to @LemonBoy 

cc @ibuclaw

### CODE REVIEW

File: src/ddmd/toctype.d [hunk 1]
```diff
@@ -170,14 +170,24 @@ public:
         //printf("TypeEnum::toCtype() '%s'\n", t.sym.toChars());
         if (t.mod == 0)
         {
+            auto baseType = t.sym.memtype.toBasetype();
             if (!t.sym.memtype)
```
Comment (ibuclaw) [inline src/ddmd/toctype.d]: Not a good place to dereference memtype, considering this condition.

**Review by ibuclaw — DISMISSED** [review]
And as per @MartinNowak comments, needs test in testsuite.

File: src/ddmd/toctype.d (hunk 1 above)
Comment (JinShil) [reply src/ddmd/toctype.d]: Ok, how about now?

Comment (JinShil) [reply src/ddmd/toctype.d]: > And as per @MartinNowak comments, needs test in testsuite.

Added.

**Comment by JinShil** [comment]
Fails for `enum E : long` on 32-bit platforms.  I don't know why yet.

**Comment by JinShil** [comment]
I'm not willing to pursue this anymore.  Closing.

**Comment by JinShil** [comment]
Filed an issue for the 32-bit backend bug at https://issues.dlang.org/show_bug.cgi?id=18191

**Comment by quickfur** [comment]
ping

Any progress on this?  This is making dmd-generated executable rather debugger-unfriendly, as enums cannot be inspected (gdb just prints `<incomplete type>` for the enum value.

**Pull request closed without merging**